singer/src/main/java/com/pinterest/singer/common/SingerSettings.java

```diff
@@ -216,7 +216,9 @@ public static void initialize(SingerConfig config)
       logMonitor = (LogMonitor) getInstanceMethod.invoke(null, monitorIntervalInSecs, singerConfig);
       logMonitor.start();
     }
-
+    long startupLatency = System.currentTimeMillis() - startTs;
+    Stats.setGauge(SingerMetrics.SINGER_STARTUP_LATENCY, startupLatency);
+    
     globalFsm.start();
```

Comment:
why not place the latency calculation after globalFsm.start();

Reply:
To prevent any future blocking calls from causing the metric to go to infinity.